rfcs/0057-named-ellipses.md

```diff
@@ -0,0 +1,36 @@
+---
+feature: Named Ellipses OR function argument consistency
+start-date: 2019-11-14
+author: deliciouslytyped
+co-authors: none
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues:
+  Named Ellipses - https://github.com/NixOS/nix/issues/2998
+---
+
+# Summary
+[summary]: #summary
+
+It should be possible to bind a name to ellipses in a function definition like `{ a, ...@extra }: null`, and `{ a, extra@... }: null`. This makes intuitive sense, and could remove the need for a lot of uses of `removeAttrs` that really just want to refer to the contents of ellipses.
+
+TODO: The latter could be substantiated.
+
+# Detailed design
+[design]: #detailed-design
+
+`{...@extraargs}: extraargs` should yield as an attrset the extra arguments "in" `...`.
+
+TODO: consider what other languages like Haskell do
+
+# Drawbacks
+[drawbacks]: #drawbacks
+None? This implements syntax that would not have worked before and so in theory shouldn't cause breakage in the Nix ecosystem. [Citation Needed]
```

Comment:
I would say that Nix has a complicated enough syntax that increasing syntax complexity should be recognised as a drawback per se to be considered in comparison to the benefit. Every syntactical construct is a cost for third-party Nix-processing tooling, for example.

(I am not saying this drawback overweights the benefit here)

Reply:
I reluctantly agree and will add something later.

Reply:
Now that I think of it, the functionality (although not the syntax) could indeed be added as a library function (as a higher-order function).